### PR TITLE
Auth VC should respect app users mode.

### PR DESCRIPTION
**Pull request by RicoYao**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @RicoYao has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m [hunk 1]
```diff
@@ -139,7 +142,14 @@ - (void)viewDidLoad
 {
     [super viewDidLoad];
 
-    [self loadAuthorizationURL];
+    if ([self.SDKClient.session isKindOfClass:[BOXOAuth2Session class]]) {
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m:145]: We couldn't do this based on the token delegate?

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m [hunk 2]
```diff
@@ -139,7 +142,14 @@ - (void)viewDidLoad
...
+        [self loadAuthorizationURL];
+    } else if ([self.SDKClient.session isKindOfClass:[BOXAppUserSession class]]) {
+        __weak BOXAuthorizationViewController *me = self;
+        [self.SDKClient autheticateAppUserWithCompletionBlock:^(BOXUser *user, NSError *error) {
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m:149]: If we're using a BOXAppUserSession, don't we know for a fact that this is what will get called by ' - (void)authenticateWithCompletionBlock:(void (^)(BOXUser *user, NSError *error))completion;' ?

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m (hunk 1 above)
Comment (RicoYao) [reply BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m:145]: I wanted to match the way we're checking this in other classes.

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m (hunk 2 above)
Comment (RicoYao) [reply BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m:149]: Discussed in-person. We're good with this current approach as is.

**Comment by boxtcarpel** [comment]
+2 LGTM

💍

**Pull request merged**